### PR TITLE
Create listing

### LINKED ISSUE #103: Create a new listing
## Description
As a registered user\
I want to create a new listing\
so that I can sell an item.

## Acceptance Criteria
- User must be logged in
- The newly created listing should be visible at the top of other listings for that category
- User must complete all mandatory fields

## Dev tasks
- [x] [Create repository for listings](https://github.com/gilhrpenner/COMP4350/issues/122)
- [x] Upload images to S3
- [ ]  [Cleanup listing endpoints and enable authentication](https://github.com/gilhrpenner/COMP4350/issues/140)
- [x] [CRUD operations for listings](https://github.com/gilhrpenner/COMP4350/issues/98)

Story points: 3/5

### PR DESCRIPTION
**Pull request by gilhrpenner**
# What I did
- Added userId to auth token so we can easily grab it on `req.user`
- Small refactoring on listing DTOs to keep them all in the same file
- Updated swagger documentation to reflect a change in the /listing endpoint
  - From what I see there is a bunch of outdated stuff
- Added enums to have a cleaner way to handle the listing category, status and condition
- Refactored frontend component to view/edit a listing
- Fixed a few tests that broke due to some of my changes
- Small refactoring on upload pictures for a listing endpoint
  - Discussed with Akash but ultimately his updates will take place to further improve those endpoints

# Todo
Some endpoints will change with Akash's PR so I just left a few // TODO: to see it easier.

Closes #103

### CODE REVIEW

File: backend/src/shared/infra/http/routes/listing.routes.ts [hunk 1]
```diff
@@ -5,26 +5,25 @@ import {
...
 
 listingRoutes.get('/:userId/:listingId', (req, res) => {
     return getListingController.handle(req, res);
 });
 
-listingRoutes.put('/:userId/:listingId', (req, res) => {
-    // TODO: protect this route
-    return putListingController.handle(req, res);
-});
+// listingRoutes.put('/:userId/:listingId', (req, res) => {
```
Comment (MarkRobitaille) [inline backend/src/shared/infra/http/routes/listing.routes.ts:23]: Is this supposed to be commented out? Just want to make sure.

Comment (gilhrpenner) [reply backend/src/shared/infra/http/routes/listing.routes.ts:23]: Yes, I discussed that with Akash, those endpoints will likely change on his PR anyways

**Review by MarkRobitaille — APPROVED** [review]
Looks good to me! 👍  

Tested it locally and it seems to work well! Once we have a section on the account settings page for managing listings, it should definitely navigate there (or at least reset the fields on the create listing page).

**Review by akashd50 — APPROVED** [review]
Looks good to me! I'll resolve the conficts in my pr as soon as this gets merged in!

**Pull request merged**